### ISSUE TITLE
User authentication using SSO and authorization

### ISSUE THREAD

**Issue by amalkurup** [AbsaOSS/spline#34]
Is user authentication using SSO, presently supported in Spline

I was also wondering, if an user is authorized to view only the lineages for spark jobs ran by him/her

**Comment by lokm01** [comment]
Hey, authentication and authorisation are definitely on our radar, but currently not supported.

Please feel free to open an issue here on github. Also any contributions are very welcome 😊

**Comment by pavanbv25** [comment]
Is there any progress on this open issues. Actually we want to showcase lineage to only authorized person of specific project. How can we achieve this. Is there any progress on oauth2 implemention. ?

**Comment by wajda** [comment]
No, we don't have any plans to implement any kind of security at the Spline core level. It can be added on top of Spline using other mechanisms, like a reverse proxy for instance.
For authorization purposes you can add some metainfo (e.g. via your custom [UserExtraMetadataProvider](https://github.com/AbsaOSS/spline-spark-agent/blob/develop/core/src/main/scala/za/co/absa/spline/harvester/extra/UserExtraMetadataProvider.scala) ) to the execution plans and use it later on the client/proxy side for access rights checking.

**Comment by wajda** [comment]
No, we don't have any plans to implement any kind of security at the Spline core level. It can be added on top of Spline using other mechanisms, like a reverse proxy for instance.
For authorization purposes you can add some metainfo (e.g. via your custom [UserExtraMetadataProvider](https://github.com/AbsaOSS/spline-spark-agent/blob/develop/core/src/main/scala/za/co/absa/spline/harvester/extra/UserExtraMetadataProvider.scala) ) to the execution plans and use it later on the client/proxy side for access rights checking.
